packages/components/src/combobox-control/stories/index.js

```diff
@@ -286,3 +286,88 @@ export const _default = CountryCodeComboboxControl.bind( {} );
 _default.args = {
 	allowReset: false,
 };
+
+const LooseMatchComboboxControl = () => {
+	const [ value, setValue ] = useState( null );
+
+	const countryStateOptions = [
```

Comment:
The `match` key is not needed anymore in this sample data object.

Also, I've added a couple of different dashes to showcase the new "dash normalization" feature

<details>

<summary>Click to expand</summary>

```diff
diff --git a/packages/components/src/combobox-control/stories/index.js b/packages/components/src/combobox-control/stories/index.js
index 5f1100a59d..11111ae609 100644
--- a/packages/components/src/combobox-control/stories/index.js
+++ b/packages/components/src/combobox-control/stories/index.js
@@ -291,56 +291,24 @@ const LooseMatchComboboxControl = () => {
 	const [ value, setValue ] = useState( null );
 
 	const countryStateOptions = [
-		{ value: 'AO:BGO', label: 'Angola — Bengo', match: 'angola bengo' },
-		{
-			value: 'AO:BLU',
-			label: 'Angola — Benguela',
-			match: 'angola — benguela',
-		},
-		{ value: 'AO:BIE', label: 'Angola — Bié', match: 'angola bié' },
-		{
-			value: 'AO:CAB',
-			label: 'Angola — Cabinda',
-			match: 'angola — cabinda',
-		},
-		{ value: 'AO:CNN', label: 'Angola — Cunene', match: 'angola cunene' },
-		{ value: 'AO:HUA', label: 'Angola — Huambo', match: 'angola huambo' },
-		{ value: 'AO:HUI', label: 'Angola — Huíla', match: 'angola huíla' },
-		{
-			value: 'AO:CCU',
-			label: 'Angola — Kuando Kubango',
-			match: 'angola kuando kubango',
-		},
-		{
-			value: 'AO:CNO',
-			label: 'Angola — Kwanza-Norte',
-			match: 'angola kwanza norte',
-		},
-		{
-			value: 'AO:CUS',
-			label: 'Angola — Kwanza-Sul',
-			match: 'angola kwanza sul',
-		},
-		{ value: 'AO:LUA', label: 'Angola — Luanda', match: 'angola luanda' },
-		{
-			value: 'AO:LNO',
-			label: 'Angola — Lunda-Norte',
-			match: 'angola lunda norte',
-		},
-		{
-			value: 'AO:LSU',
-			label: 'Angola — Lunda-Sul',
-			match: 'angola lunda sul',
-		},
-		{
-			value: 'AO:MAL',
-			label: 'Angola — Malanje',
-			match: 'angola malanje',
-		},
-		{ value: 'AO:MOX', label: 'Angola — Moxico', match: 'angola moxico' },
-		{ value: 'AO:NAM', label: 'Angola — Namibe', match: 'angola namibe' },
-		{ value: 'AO:UIG', label: 'Angola — Uíge', match: 'angola uíge' },
-		{ value: 'AO:ZAI', label: 'Angola — Zaire', match: 'angola zaire' },
+		{ value: 'AO:BGO', label: 'Angola ~ Bengo' },
+		{ value: 'AO:BLU', label: 'Angola - Benguela' },
+		{ value: 'AO:BIE', label: 'Angola – Bié' },
+		{ value: 'AO:CAB', label: 'Angola — Cabinda' },
+		{ value: 'AO:CNN', label: 'Angola — Cunene' },
+		{ value: 'AO:HUA', label: 'Angola — Huambo' },
+		{ value: 'AO:HUI', label: 'Angola — Huíla' },
+		{ value: 'AO:CCU', label: 'Angola — Kuando Kubango' },
+		{ value: 'AO:CNO', label: 'Angola — Kwanza-Norte' },
+		{ value: 'AO:CUS', label: 'Angola — Kwanza-Sul' },
+		{ value: 'AO:LUA', label: 'Angola — Luanda' },
+		{ value: 'AO:LNO', label: 'Angola — Lunda-Norte' },
+		{ value: 'AO:LSU', label: 'Angola — Lunda-Sul' },
+		{ value: 'AO:MAL', label: 'Angola — Malanje' },
+		{ value: 'AO:MOX', label: 'Angola — Moxico' },
+		{ value: 'AO:NAM', label: 'Angola — Namibe' },
+		{ value: 'AO:UIG', label: 'Angola — Uíge' },
+		{ value: 'AO:ZAI', label: 'Angola — Zaire' },
 	];
 
 	return (
```

</details>